content/docs/reference/cli-branches.md

```diff
@@ -0,0 +1,198 @@
+| --endpoint.type                           | The compute endpoint type. Either `read_write` or `read_only`. The `read_only` compute endpoint type is not yet supported. The default value is `read_write`. | string  |              |
+| --endpoint.provisioner                    | The Neon compute provisioner. Supported values are `k8s-pod` and `k8s-neonvm`. The later is required to use _Autoscaling_.                                                               | string  |      |
+| --endpoint.suspend_timeout_seconds        | Duration of inactivity in seconds after which endpoint will be automatically suspended. Value `0` means use global default, `-1` means never suspend. The maximum value is 604800 seconds (1 week).
+
+#### Example
+
+```bash
+neonctl branches create --project.id spring-sky-578180
+┌─────────────────────────┬─────────────────────────┬──────────────────────┐
+│ Id                      │ Name                    │ Created At           │
+├─────────────────────────┼─────────────────────────┼──────────────────────┤
+│ br-withered-king-763176 │ br-withered-king-763176 │ 2023-06-19T22:35:25Z │
+└─────────────────────────┴─────────────────────────┴──────────────────────┘
+```
+
+### update
+
+This subcommand allows you to update a branch in a Neon project.
+
+#### Usage
+
+```bash
+neonctl branches update [options]
+```
+
+#### Options
+
+In addition to the Neon CLI [global options](../neon-cli/global-options), the `update` subcommand supports these options:
+
+| Option        | Description | Type   | Required  |
+| ------------- | ----------- | ------ | :-----: |
+| --project.id  | Project ID  | string | &check; |
+| --branch.name |             | string |         |
+| --branch.id   | Branch ID   | string | &check; |
+
+#### Example
+
+```bash
+neonctl branches update --project.id spring-sky-578180 --branch.id br-withered-king-763176 --branch.name mynewbranch
+┌─────────────────────────┬─────────────┬──────────────────────┐
+│ Id                      │ Name        │ Created At           │
+├─────────────────────────┼─────────────┼──────────────────────┤
+│ br-withered-king-763176 │ mynewbranch │ 2023-06-19T22:35:25Z │
+└─────────────────────────┴─────────────┴──────────────────────┘
+```
+
+### delete
+
+This subcommand allows you to delete a branch in a Neon project.
+
+#### Usage
+
+```bash
+neonctl branches delete [options]
+```
+
+#### Options
+
+In addition to the Neon CLI [global options](../neon-cli/global-options), the `delete` subcommand supports these options:
+
+| Option        | Description | Type   | Required  |
+| ------------- | ----------- | ------ | :------: |
+| --project.id  | Project ID  | string | &check; |
+| --branch.id   | Branch ID   | string | &check; |
```

Comment:
@duskpoet , can we also add the support of deleting branches by name?
we have this in the API too, this is much more friendly for the users IMO

Reply:
Where did you find that api? I don't see anything like this in docs

Reply:
Updated in the docs as per the latest CLI